### PR TITLE
Add logic to handle patch requests that reference the root...

### PR DESCRIPTION
**Pull request by awoods**
- Removing "jcr:content" from authorization checks

Partial resolution of: https://jira.duraspace.org/browse/FCREPO-1725

### CODE REVIEW

File: fcrepo-auth-common/src/main/java/org/fcrepo/auth/common/FedoraUserSecurityContext.java [hunk 1]
```diff
@@ -37,6 +37,9 @@
     private static final Logger LOGGER = LoggerFactory
             .getLogger(FedoraUserSecurityContext.class);
 
+    // Defined here because FedoraJcrTypes is in fcrepo-http-api, which would create a cyclic dependency
+    protected static final String JCR_CONTENT = "jcr:content";
```
Comment (acoburn) [inline fcrepo-auth-common/src/main/java/org/fcrepo/auth/common/FedoraUserSecurityContext.java]: `FedoraJcrTypes` is in fcrepo-kernel-api, and if `jcr:content` were defined there, I'm not sure that I see how a cyclic dependency would be created.

Comment (awoods) [inline fcrepo-auth-common/src/main/java/org/fcrepo/auth/common/FedoraUserSecurityContext.java]: Try it, it is true.

Comment (acoburn) [inline fcrepo-auth-common/src/main/java/org/fcrepo/auth/common/FedoraUserSecurityContext.java]: I disagree. See my pull request: https://github.com/awoods/fcrepo4/pull/4

Comment (awoods) [inline fcrepo-auth-common/src/main/java/org/fcrepo/auth/common/FedoraUserSecurityContext.java]: Thanks for the PR.
Would you agree that your PR creates a dependency from fcrepo-auth-common on to fcrepo-http-api? This new dependency is coming transitively from some other artifact... but exists, nonetheless. If you explicitly add the dependency into fcrepo-auth-common's pom.xml, Maven complains.

Are you comfortable relying on this transitive coupling?
Am I missing something?

Comment (acoburn) [inline fcrepo-auth-common/src/main/java/org/fcrepo/auth/common/FedoraUserSecurityContext.java]: We're talking about different things.

This PR creates a dependency on `fcrepo-kernel-api`, not `fcrepo-http-api`. If the dependency to `fcrepo-kernel-api` is added to maven (which it should be: see my additional commit), there is no complaint. This was the whole point, a few weeks ago, in removing `fcrepo-mint` from the kernel-api: that way the kernel-api is a the very bottom of food-chain, as it were, and *any* other module should freely be able to use it.

Comment (ajs6f) [inline fcrepo-auth-common/src/main/java/org/fcrepo/auth/common/FedoraUserSecurityContext.java]: Yes, if there is any problem with any module at all using the API, that is a massive architecture failure. the API should be essentially independent of _everything else_. I'm not saying it is, I'm saying that such is the condition for which we must be working.

Comment (awoods) [inline fcrepo-auth-common/src/main/java/org/fcrepo/auth/common/FedoraUserSecurityContext.java]: Thanks, @acoburn. You are correct, I was transposing `fcrepo-http-api` for `fcrepo-kernel-api`.

**Comment by acoburn** [comment]
Merged via 249977cae0036673fe2e293097380947c7052f63

**Pull request closed without merging**